it/controllers/benefits/CompanyVanFuelBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,438 @@
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the 'Did you get fuel benefit for a company van?' page with correct content and yes button selected when the user " +
+          "has previously chosen yes but has did not enter a vanFuel amount yet" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+            insertCyaData(cya(isPriorSubmission = false, benefitsWithVanFuelYes(vanFuelAmount = None)), User(mtditid, None, nino, sessionId, affinityGroup))
+            urlGet(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedH1)
+          textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, Some(true))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueButtonLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the 'Did you get fuel benefit for a company van?' page with correct content and no button selected when there cya" +
+          "data for the question set as false" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+            insertCyaData(cya(isPriorSubmission = false, benefitsWithVanFuelNo), User(mtditid, None, nino, sessionId, affinityGroup))
+            urlGet(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedH1)
+          textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, Some(false))
+          radioButtonCheck(noText, 2, Some(true))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueButtonLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "redirect to overview page if the user tries to hit this page with current taxYear" when {
```

Comment:
This is a requirement off the acceptance criteria so this same test should also be done in the submit section which is missing at the moment